src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1474,6 +1501,24 @@ void DoCancelRequest(int backendProcessId, int backendSecretKey)
             }
         }
 
+        internal void ResetCancellation()
+        {
+            _cancellationRequested = false;
+            _userCancellationRequested = false;
+
+            if (ReadCts.IsCancellationRequested)
+            {
+                ReadCts.Dispose();
+                ReadCts = new TimeoutCancellationTokenSourceWrapper(TimeSpan.FromSeconds(Settings.CancellationTimeout));
+            }
+
+            if (WriteCts.IsCancellationRequested)
```

Comment:
Since cancelling writing means breaking the connector, under what circumstances are we expecting to need a new WriteCts here? If we really need to do this, any reason not to use the new wrapper?